minimal_scheduling/minimal_sched_fifo_callback_group.cpp

```diff
@@ -0,0 +1,152 @@
+  {
+    publisher1_ = this->create_publisher<std_msgs::msg::String>("topic", 10);
+    auto timer_callback1 =
+      [this]() -> void {
+        auto message = std_msgs::msg::String();
+        message.data = "Hello, world! " + std::to_string(this->count1_++);
+        RCLCPP_INFO(this->get_logger(), "Publishing     : '%s'", message.data.c_str());
+        this->publisher1_->publish(message);
+      };
+    timer1_ = this->create_wall_timer(500ms, timer_callback1);
+
+    realtime_callback_group_ = this->create_callback_group(
+      rclcpp::CallbackGroupType::MutuallyExclusive, false);
+
+    publisher2_ = this->create_publisher<std_msgs::msg::String>("topic_rt", 10);
+    auto timer_callback2 =
+      [this]() -> void {
+        auto message = std_msgs::msg::String();
+        message.data = "Hello, world! " + std::to_string(this->count2_++);
+        RCLCPP_INFO(this->get_logger(), "Publishing (RT): '%s'", message.data.c_str());
+        this->publisher2_->publish(message);
+      };
+    timer2_ = this->create_wall_timer(500ms, timer_callback2, realtime_callback_group_);
+  }
+
+  rclcpp::CallbackGroup::SharedPtr get_realtime_callback_group()
+  {
+    return realtime_callback_group_;
+  }
+
+private:
+  rclcpp::TimerBase::SharedPtr timer1_, timer2_;
+  rclcpp::Publisher<std_msgs::msg::String>::SharedPtr publisher1_, publisher2_;
+  size_t count1_, count2_;
+  rclcpp::CallbackGroup::SharedPtr realtime_callback_group_;
+};
+
+class MinimalSubscriber : public rclcpp::Node
+{
+public:
+  MinimalSubscriber()
+  : Node("minimal_subscriber")
+  {
+    subscription1_ = this->create_subscription<std_msgs::msg::String>(
+      "topic",
+      10,
+      [this](std_msgs::msg::String::UniquePtr msg) {
+        RCLCPP_INFO(this->get_logger(), "I heard:        '%s'", msg->data.c_str());
+      });
+
+    realtime_callback_group_ = this->create_callback_group(
+      rclcpp::CallbackGroupType::MutuallyExclusive, false);
+
+    rclcpp::SubscriptionOptions subscription_options;
+    subscription_options.callback_group = realtime_callback_group_;
+    subscription2_ = this->create_subscription<std_msgs::msg::String>(
+      "topic_rt",
+      10,
+      [this](std_msgs::msg::String::UniquePtr msg) {
+        RCLCPP_INFO(this->get_logger(), "I heard (RT):   '%s'", msg->data.c_str());
+      }, subscription_options);
+  }
+
+  rclcpp::CallbackGroup::SharedPtr get_realtime_callback_group()
+  {
+    return realtime_callback_group_;
+  }
+
+private:
+  rclcpp::Subscription<std_msgs::msg::String>::SharedPtr subscription1_;
+  rclcpp::Subscription<std_msgs::msg::String>::SharedPtr subscription2_;
+  rclcpp::CallbackGroup::SharedPtr realtime_callback_group_;
+};
+
+
+int main(int argc, char * argv[])
+{
+  rclcpp::init(argc, argv);
+
+  auto node_pub = std::make_shared<MinimalPublisher>();
+  auto node_sub = std::make_shared<MinimalSubscriber>();
+
+  rclcpp::executors::StaticSingleThreadedExecutor default_callback_group_executor;
```

Comment:
For readability, I would name it simply:

```suggestion
  rclcpp::executors::StaticSingleThreadedExecutor default_executor;
```